src/createproject.ts

```diff
@@ -0,0 +1,157 @@
+import * as fs from 'async-file'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { JuliaExecutablesFeature } from './juliaexepath'
+import { registerCommand } from './utils'
+
+export class JuliaNewProjectFeature {
+    constructor(private context: vscode.ExtensionContext, private juliaExecutablesFeature: JuliaExecutablesFeature) {
+        this.context.subscriptions.push(registerCommand('language-julia.createNewProject', () => this.createNewProject()))
+    }
+
+    private async createNewProject() {
+
+        const pkgName = await vscode.window.showInputBox({
+            prompt: 'Please enter the name of the project to create.',
+            validateInput: (input) => {
+                if (input === "")
+                    return 'The project name cannot be empty.'
+                return undefined
+            }
+        })
+        if (!pkgName)
+            return
+
+        const authors = await vscode.window.showInputBox({ prompt: 'Please enter the authors of the project', placeHolder: "Default uses 'github.name' and 'github.email' from the global Git config." })
+        if (authors === undefined)
+            return
+
+        let host = await vscode.window.showQuickPick(['github.com', 'gitlab.com', 'bitbucket.org', 'Other'], { placeHolder: 'The URL to the code hosting service where the project will reside.' })
+        if (host === 'Other') {
+            host = await vscode.window.showInputBox({ prompt: 'Please enter the URL to the code hosting service.' })
+        }
+        if (host === undefined)
+            return
+
+        // TODO: add gihub.user as placeholder
+        const user = await vscode.window.showInputBox({
```

Comment:
Can we also get that from the built-in VS Code authentication stuff?

Reply:
I found that you can access the global git config through vscode's git api but it requires a repository to exist already which doesn't make a lot of sense here.
```ts
const extension = vscode.extensions.getExtension<GitExtension>('vscode.git')
if (extension == null && extension.isActive) {
    const gitExtension = extension.exports
    const api = gitExtension?.getAPI(1)
    const user = await api.repositories[0].getGlobalConfig('github.user')
}
```
There's [simple-git](https://www.npmjs.com/package/simple-git) that does the job. Does it make sense to add a dependency?